samples/ingestion/ingestion-client/StartTranscriptionByServiceBus/Startup.cs

```diff
@@ -0,0 +1,57 @@
+// <copyright file="Startup.cs" company="Microsoft Corporation">
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
+// </copyright>
+
+using Microsoft.Azure.Functions.Extensions.DependencyInjection;
+
+[assembly: FunctionsStartup(typeof(StartTranscriptionByServiceBus.Startup))]
+
+namespace StartTranscriptionByServiceBus
+{
+    using System;
+
+    using Azure.Storage;
+    using Azure.Storage.Blobs;
+
+    using Connector;
+
+    using Microsoft.Azure.Functions.Extensions.DependencyInjection;
+    using Microsoft.Extensions.DependencyInjection;
+
+    using StartTranscriptionByTimer;
+
+    public class Startup : FunctionsStartup
+    {
+        public override void Configure(IFunctionsHostBuilder builder)
+        {
+            _ = builder ?? throw new ArgumentNullException(nameof(builder));
+
+            var storageConnectionString = StartTranscriptionEnvironmentVariables.AzureWebJobsStorage;
+            var blobServiceClient = new BlobServiceClient(storageConnectionString);
+            var storageCredential = new StorageSharedKeyCredential(
+                GetValueFromConnectionString("AccountName", storageConnectionString),
+                GetValueFromConnectionString("AccountKey", storageConnectionString));
+
+            builder.Services.AddSingleton(blobServiceClient);
+            builder.Services.AddSingleton(storageCredential);
+
+            builder.Services.AddTransient<IStorageConnector, StorageConnector>();
```

Comment:
@komalg1 could you explain the significance of creating this as a transient instance as opposed to a singleton like blobServiceClient and storageCredential?

Reply:
Singleton service is created once & can be shared safely across multiple threads. Basically it prevents overhead of creating same objects. In this case for both BlobServiceClient & StorageSharedKeyCredential, we don't want to repeatedly create & dispose blob client involving recreating the connections (using credentials) every time.
Transient service is created each time it is requested. Mostly for services which hold context or state specific to the operation/request. So for StorageConnector, it might hold some context specific to the request. Basically it will instantiate a new instance every time & therefore prevent any clash in case of concurrent requests & any internal state or config will not be shared.
 
Also as a general case, Singleton classes are efficient for usage in case of connections & secret management because we don't want to recreate the same connection objects.